extensions/ql-vscode/src/variant-analysis/gh-api/gh-api-client.ts

```diff
@@ -20,18 +23,38 @@ export async function getCodeSearchRepositories(
   token: CancellationToken,
 ): Promise<string[]> {
   let nwos: string[] = [];
-  const octokit = await credentials.getOctokit();
+  const MyOctokit = Octokit.plugin(throttling);
+  const auth = await credentials.getAccessToken();
+
+  const octokit = new MyOctokit({
+    auth,
+    throttle: {
+      onRateLimit: (retryAfter: number, options: any): boolean => {
+        void showAndLogWarningMessage(
+          `Request quota exhausted for request ${options.method} ${options.url}. Retrying after ${retryAfter} seconds!`,
+        );
+
+        return true;
+      },
+      onSecondaryRateLimit: (_retryAfter: number, options: any): void => {
+        void showAndLogWarningMessage(
+          `SecondaryRateLimit detected for request ${options.method} ${options.url}`,
```

Comment:
```suggestion
          `Request quota exceeded for request ${options.method} ${options.url}`.,
```
Just a suggestion to make this error fit with the other one.

Reply:
I think it makes sense to distinguish between the two, so that users know what they are hitting specifically.

Reply:
I agree users should be able to distinguish and understand them. I was meaning more that we could make the language style more consistent between the two error messages. For example:
- In one we say "quota" and in the other we say "limit". Is there a technical different between these I'm not aware of, and will the user know what they mean?
- In one we use full sentences, and in the other we use CamelCase.

However, this is something that can easily be changed afterwards, if time is tight. So it's ok to defer this to a later PR if needed.

Reply:
I took this from the documentation and it copies also a script we have somewhere else in the repo. However you're right with the CamelCase, that's just weird. I fixed the CC and adjusted the wording!